frontend/packages/dev-console/src/components/topology/__tests__/topology-page.spec.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import { StatusBox, LoadingBox } from '@console/internal/components/utils';
+import Topology from '../Topology';
+import { TopologyPage, RenderTopology } from '../TopologyPage';
+import ProjectsExistWrapper from '../../ProjectsExistWrapper';
+import { topologyData } from './topology-test-data';
+
+type TopologyPageProps = React.ComponentProps<typeof TopologyPage>;
+type RenderTopologyProps = React.ComponentProps<typeof RenderTopology>;
+
+let topologyProps: TopologyPageProps;
+let renderTopologyProps: RenderTopologyProps;
+
+describe('Topology page tests', () => {
+  beforeEach(() => {
+    topologyProps = {
+      activeApplication: 'topology-test',
+      knative: false,
+      cheURL: 'che.openshift.io',
+      serviceBinding: false,
+      match: {
+        params: {
+          name: 'topology-test',
+        },
+        isExact: true,
+        path: '',
+        url: '',
+      },
+    };
+
+    renderTopologyProps = {
+      data: topologyData,
+      loaded: true,
+      loadError: '',
+      serviceBinding: false,
+    };
+  });
+
+  it('should render topology page', () => {
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    expect(wrapper.exists()).toBe(true);
+    expect(wrapper.contains(<title>Topology</title>)).toBe(true);
+  });
+
+  it('should render topology list page', () => {
+    topologyProps.match.path = '/list';
+    topologyProps.match.params.name = '';
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    expect(typeof wrapper.find(ProjectsExistWrapper).props().children).toBe('function');
+    const projectListWrapper = shallow(
+      wrapper
+        .find(ProjectsExistWrapper)
+        .props()
+        .children(),
+    );
+    expect(projectListWrapper.find('.odc-project-list-page').exists()).toBe(true);
+  });
+
+  it('should render render topology test', () => {
+    const wrapper = shallow(<RenderTopology {...renderTopologyProps} />);
+    expect(wrapper.find(StatusBox).exists()).toBe(true);
+    expect(wrapper.find('.odc-topology').exists()).toBe(true);
+    expect(wrapper.find(Topology).exists()).toBe(true);
+  });
+
+  it('should render loading box if data not loaded', () => {
+    renderTopologyProps.loaded = false;
+    const wrapper = shallow(<RenderTopology {...renderTopologyProps} />);
+    const loadingBoxWrapper = wrapper.find(StatusBox).shallow();
+    expect(loadingBoxWrapper.find(LoadingBox).exists()).toBe(true);
+  });
```

Comment:
Need a test to check for Project List page when `all projects` is selected in the namespace

Reply:
I already added this use case. see here https://github.com/openshift/console/pull/3829/files#diff-3ff93205d605f09dfbf10bf6adf7dfcbR50